src/languages/rust.ts

```diff
@@ -0,0 +1,75 @@
+import { SimpleScopeTypeType } from "../typings/targetDescriptor.types";
+import { NodeMatcherAlternative } from "../typings/Types";
+import { patternFinder } from "../util/nodeFinders";
+import {
+  ancestorChainNodeMatcher,
+  argumentMatcher,
+  cascadingMatcher,
+  createPatternMatchers,
+  leadingMatcher,
+  patternMatcher,
+  trailingMatcher,
+} from "../util/nodeMatchers";
+
+// Generated by the following command:
+// `curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-rust/36ae187ed6dd3803a8a89dbb54f3124c8ee74662/src/node-types.json | jq '[.[] | select(.type == "_declaration_statement") | .subtypes[].type, "expression_statement"]'`
+const STATEMENT_TYPES = [
+  "associated_type",
+  "attribute_item",
+  "const_item",
+  "empty_statement",
+  "enum_item",
+  "extern_crate_declaration",
+  "foreign_mod_item",
+  "impl_item",
+  "inner_attribute_item",
+  "let_declaration",
+  "macro_definition",
+  "macro_invocation",
+  "mod_item",
+  "static_item",
+  "struct_item",
+  "trait_item",
+  "type_item",
+  "union_item",
+  "use_declaration",
+  "expression_statement",
+];
+
+/**
+ * Scope types allowed to be parents of a statement
+ */
+const STATEMENT_PARENT_TYPES = ["source_file", "block", "declaration_list"];
+
+const nodeMatchers: Partial<
+  Record<SimpleScopeTypeType, NodeMatcherAlternative>
+> = {
+  statement: STATEMENT_TYPES,
+  string: ["raw_string_literal", "string_literal"],
+  ifStatement: "if_expression",
+  functionCall: ["call_expression", "macro_invocation", "struct_expression"],
+  comment: ["line_comment", "block_comment"],
+  list: ["array_expression", "tuple_expression"],
+  collectionItem: argumentMatcher("array_expression", "tuple_expression"),
+  namedFunction: "function_item",
+  type: cascadingMatcher(
+    leadingMatcher(
+      ["let_declaration[type]", "parameter[type]", "field_declaration[type]"],
+      [":"]
+    ),
+    leadingMatcher(["function_item[return_type]"], ["->"]),
+    patternMatcher("struct_item", "trait_item", "impl_item", "impl_item[type]")
+  ),
+  functionName: ["function_item[name]"],
+  anonymousFunction: "closure_expression",
+  argumentOrParameter: cascadingMatcher(
+    argumentMatcher("arguments"),
+    trailingMatcher(["parameter"], [","]),
+    trailingMatcher(["self_parameter"], [","])
```

Comment:
I don't think this will work in all cases.  I believe it will break if you say "chuck arg" on the final parameter if there is no trailing comma.  In that case, it is not a trailing `,` that you want to delete, but leading.  That's why we always use `argumentMatcher` for comma-separated lists.  Admittedly that name could be clearer, to indicate it's for all comma-separated lists 😅.  Maybe you'll need to tweak `argumentMatcher` to support specifying child type rather than parent type?